### PR TITLE
Add filtering to repo selection, and the option to chose an org instead of personal repos

### PR DESCRIPTION
**Pull request by tiefpunkt**
Previously, all repos the user has access to were selected, and then 
they were dismissed later on, which led to much output, and longer 
runtimes. Now, repos are selcted based on visibility and ownership when 
getting the repo list.

Also, you can now specify an organization to modify instead of your 
personal repositories.

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -28,24 +29,28 @@ def write_topics(repo, topiclist, dry_run):
 g = Github(token)
 
 try:
-    user_repositories = list(g.get_user().get_repos())
+    print("Getting list of repositories")
+    if args.organization:
+        repositories = [r for r in g.get_user().get_repos(affiliation = "organization_member", visibility = "public") if r.organization.login == args.organization]
```
Comment (fleaz) [inline run.py:34]: Won't this cause permission errors for organizations where I'm a member but don't have the admin rights to set topics?

File: run.py [hunk 2]
```diff
@@ -28,24 +29,28 @@ def write_topics(repo, topiclist, dry_run):
...
     user_id = g.get_user().id
 except BadCredentialsException:
     print("Token is invalid or has the wrong permissions")
     exit(1)
 
-for repo in user_repositories:
+for repo in repositories:
     print(f"> Checking {repo.full_name}:")
 
     if repo.private or repo.fork or repo.archived:
```
Comment (fleaz) [inline run.py]: At least the `repo.private` check shouldn't be neccessary anymore due to the new filters in `get_repos()` I guess?

File: run.py [hunk 3]
```diff
@@ -28,24 +29,28 @@ def write_topics(repo, topiclist, dry_run):
...
         print("Repo is private, a fork or archived. Skipping...")
         continue
 
-    if repo.organization is not None:
+    if not args.organization and repo.organization is not None:
         print("Repo belongs to a organization. Skipping...")
         continue
 
-    if repo.owner.id != user_id:
+    if not args.organization and repo.owner.id != user_id:
```
Comment (fleaz) [inline run.py]: Can this rvrn happen If we filter for `affiliation = "owner"`?

**Review by fleaz — CHANGES_REQUESTED** [review]

File: run.py (hunk 1 above)
Comment (tiefpunkt) [reply run.py:34]: It propably does, but I'm not sure how to check for those permissions. There's https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-repository-permissions-for-a-user and also a permission dict in the repo response, but I couldn't find any information on the necessary permissions to set topics. We could just check for the admin permission, which is likely what it is...

File: run.py (hunk 2 above)
Comment (tiefpunkt) [reply run.py]: Yes, I'll remove that one.

File: run.py (hunk 3 above)
Comment (tiefpunkt) [reply run.py]: I'd assume so. Let's give it a shot.

File: run.py [hunk 4]
```diff
@@ -28,24 +29,28 @@ def write_topics(repo, topiclist, dry_run):
...
-for repo in user_repositories:
+for repo in repositories:
     print(f"> Checking {repo.full_name}:")
 
     if repo.private or repo.fork or repo.archived:
         print("Repo is private, a fork or archived. Skipping...")
         continue
 
-    if repo.organization is not None:
+    if not args.organization and repo.organization is not None:
```
Comment (tiefpunkt) [inline run.py:49]: Thinking about it, shouldn't this also be obsolete with the `affiliation = "owner"` filter? @fleaz

File: run.py (hunk 1 above)
Comment (tiefpunkt) [reply run.py:34]: "Repository admins can add any topics they'd like to a repository." (https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/classifying-your-repository-with-topics#about-topics) - let's just check for the admin permission

**Review by fleaz — APPROVED** [review]

**Comment by fleaz** [comment]
Thanks for the work  :+1:

**Pull request merged**